From 304bf870669afc9621d52ea1e446af345ef93bc5 Mon Sep 17 00:00:00 2001
From: Jordan Paige Hendricks <jordan.hendricks@joyent.com>
Date: Tue, 13 Sep 2016 22:55:27 +0000
Subject: [PATCH] MANTA-2908 muskie logs misleading about remoteIp and
 remotePort Reviewed by: Dave Pacheco <dap@joyent.com> Approved by: Dave
 Pacheco <dap@joyent.com>

---
 lib/audit.js | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/lib/audit.js b/lib/audit.js
index 50582d4..36bfa92 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -209,7 +209,13 @@ function auditLogger(options) {
             operation: name,
             billable_operation: op,
             bytesTransferred: req._size,
-            remoteAddress: req.connection._xff,
+            /*
+             * If the request has an X-Forwarded-For header from haproxy,
+             * logicalRemoteAddress will be the logical source IP of the client.
+             * Otherwise, it will be the same as remoteAddress.
+             */
+            logicalRemoteAddress: req.connection._xff,
+            remoteAddress: req.connection.remoteAddress,
             remotePort: req.connection.remotePort,
             req_id: req.id,
             reqHeaderLength: reqHeaderLength,
-- 
2.21.0

